cmake/kokkos_arch.cmake

```diff
@@ -48,6 +48,8 @@ declare_and_check_host_arch(NATIVE "local machine")
 declare_and_check_host_arch(AMDAVX "AMD chip")
 declare_and_check_host_arch(ARMV80 "ARMv8.0 Compatible CPU")
 declare_and_check_host_arch(ARMV81 "ARMv8.1 Compatible CPU")
+declare_and_check_host_arch(ARMV84 "ARMv8.4 Compatible CPU")
+declare_and_check_host_arch(ARMV84_SVE "Generic ARMv8.4 with SVE support (-march=armv8.4-a+sve)")
```

Comment:
Given that this is basically the architecture of a Neoverse V1-based CPU, it may be simpler to define this as `NEOVERSE_V1`, i.e., use a specific Arm _microarchitecture_ instead of a generic architecture. It is kind of what Kokkos already does with, e.g., A64FX (which is its own microarchitecture implementing Armv8.2-A+SVE). As far as I know, this would also better match SiPearl's Rhea1 (and AWS Graviton3).

Reply:
Honestly, I'm not keen on setting arch-specific CPU option for such a SVE ISA, which was aimed to be generic and portable. It would introduce further CPU-options bloating in Kokkos. We can already see it with `KOKKOS_ARCH_A64FX` and `KOKKOS_ARCH_ARMV9_GRACE`, which can typically use the same SVE backend implementation, with just different VL-fixed sizes. We can also imagine the scenario with future RISC-V architectures & vendors.

BTW, it remains my personal view, I'd like to hear other opinions on this "fragmentation-prone" aspect.

Reply:
Perhaps in the future we can have a look at https://github.com/archspec and try to be coherent with their model.